### DISCUSSION TITLE
[Typescript] How to pass the type for the error using useQuery?

### DISCUSSION THREAD

**Discussion by cant89** [TanStack/query#934] [General] [answered]
Hi, 

I can't find a way to infer the type for the error of the `useQuery` response, it always comes as `unknown`.
Is there a way? 
If not, could you add the possibility to pass it to `type QueryResult` as with the `TResult`?

Thanks

**Comment by alinjie** [comment, marked as answer]
Hi!

The error type can be set like this


```typescript
interface Todo {
  userId: number;
  id: number;
  title: string;
  completed: boolean;
}

function useTodos(): QueryResult<Todo[], Error> {
  return useQuery("todos", async () => {
    const response = await fetch("https://jsonplaceholder.typicode.com/todos");

    if (!response.ok) {
      throw new Error("Failed to fetch!");
    }

    return await response.json();
  });
}
```

If the function throws the `error` variable will be of type `Error`, as specified in `QueryResult`.

Hope this helps!

**Reply by cant89** [reply]
Ops sorry, I just had to update to the latest version. Thanks for the quick answer.

**Reply by MAb540** [reply]
for latest version of react query use **UseQueryResult<Todo[], Error>**.

**Comment by diegoarcega** [comment]
The second argument in the `useQuery` generic is the type of the error. 


what I did was:

```typescript

interface App {} // response in "data"
interface ApiError {} // error object

const { data, error } = useQuery<App, ApiError>(['app', id], () =>
    AppsService.getApiV1Apps(id)
  );



```

**Comment by TkDodo** [comment]
for v5, please see:

https://tanstack.com/query/v5/docs/react/typescript#registering-a-global-error
